File: LP2/isprime.c
#include "stdio.h"
#include "stdbool.h"

bool isPrime(int number);

int main(void)
{
    int num;

    printf("Digite um número: ");
    scanf("%d", &num);

    char* res = isPrime(num) ? "%d é primo!\n" : "%d não é primo...\n";

    printf(res, num);

    return 0;
}

bool isPrime(int number)
{
    if (number < 2)
    {
        return false;
    }

    for (int i = 2; i < number; i++)
    {
        if (number % i == 0)
        {
            return false;
        }
    }

    return true;
}

File: LP/GO/isPrime.go
package main

import (
	"fmt"
)

func isPrime(number int) bool {
	if number < 2 {
		return false
	}
	for i := 2; i < number; i++ {
		if number%i == 0 {
			return false
		}
	}
	return true
}

func main() {
	var num int
	fmt.Print("Digite um número: ")
	fmt.Scan(&num)

	if isPrime(num) {
		fmt.Printf("%d é primo!\n", num)
	} else {
		fmt.Printf("%d não é primo...\n", num)
	}
}
